src/north/amazon/AmazonS3.class.spec.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs')
+const path = require('path')
+const ApiHandler = require('../ApiHandler.class')
+const AmazonS3 = require('./AmazonS3.class')
+// const config = require('../../config/defaultConfig.json')
+const config = require('../../../tests/testConfig').default
+
+
+// Mock database service 
+jest.mock('../../services/database.service',() => {})
+
+// Mock logger
+jest.mock('../../engine/Logger.class')
+
+// Mock engine
+const engine = jest.genMockFromModule('../../engine/Engine.class')
+engine.configService = { getConfig: () => ({ engineConfig: config.engine }) }
+
+beforeEach(() => {
+jest.resetAllMocks()
+})
+
+describe('Amazone S3 north', () => {
+
+  it('should be properly initialized', () => {
+    const AmazonS3Config = config.north.applications[8]
+    const AmazonS3North = new AmazonS3(AmazonS3Config, engine)
+    expect(AmazonS3North.canHandleFiles).toBeTruthy()
+    // expect() tobeContinued
```

Comment:
You can check if methods called from the constructor are called with the expected parameters. You will need to mock the different methods

Reply:
expect(AmazonS3North.bucket).toBe(AmazonS3Config.AmazonS3.bucket)